### ISSUE TITLE
Breakpoint 'teleports' when used as the last statement within a context block 

### ISSUE THREAD

**Issue by JosephSBoyle** [python/cpython#98965]
# Bug report

`breakpoint` statements in the last line of a context manager block do not behave as expected, they instead `breakpoint` into the enclosing context's `__exit__` (or `__aexit__`, for `async` context managers) method.

```python
class Context:
    def __enter__(self):
        pass
    
    def __exit__(self, *args, **kwargs):
        pass

def test():
    r = Context()

    with Context():
        foo = None
        breakpoint()
        # > ../home/<username>/tests/unit/test_breakpoint_error.py(5)__exit__()
        # -> def __exit__(self, *args, **kwargs):
        # > foo
        # foo is not defined!
        # > dir()
        # ['args', 'kwargs', 'self']
        # > self
        # <...Context object at 0x7f36756494c0>
```

Note that if we add a statement after the breakpoint, still inside the context like so:

```python
   with Context():
        foo = None
        breakpoint()
        bar = foo
```
Then the breakpoint works as expected.

# Your environment

- Python 3.9.4 (default, Mar 28 2022, 14:17:01) 
[GCC 9.4.0] on linux

- Ubuntu 20.04.4 LTS

**Comment by sweeneyde** [comment]
It looks like this was fixed in this commit:

> 5979e81a212949c62c2490167c9137d233d7de64 is the first new commit
> commit 5979e81a212949c62c2490167c9137d233d7de64
> Author: Mark Shannon <mark@hotpy.org>
> Date:   Fri Apr 30 14:32:47 2021 +0100
> 
>     bpo-43933:  Set frame.f_lineno during call to __exit__ (GH-25719)
> 
>     * Set line number of __exit__ call in a with statement to be that of the with keyword.

Python 3.9 is only accepting security-related fixes, but the issue is fixed in 3.10 and 3.11.

**Comment by JosephSBoyle** [comment]
@sweeneyde In that case I will close this issue :+1:.
